### PR TITLE
Bug/300 configure transaction timeout

### LINKED ISSUE #300: Transaction timeouts should be configurable
The amount of time Packetbeat waits for a response should be configurable. Currently the timeout value for most protocols is hardcoded to 10 seconds. If a service takes longer than 10 seconds to respond then the response is not logged.

https://discuss.elastic.co/t/timeout-connection-are-not-being-monitored-by-packetbeat/31581

### PR DESCRIPTION
**Pull request by urso**
configure per protocol transaction timeout. This PR also updates docs.

Requires elastic/libbeat#144 to be merged first + godep update of libbeat to master after merging.

Fixes #300

### CODE REVIEW

**Comment by urso** [comment]
PR contains libbeat update to recent master

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -403,8 +407,8 @@ fields.
     parseunknown: false
     maxvalues: 0
     maxbytespervalue: 100
+    transaction-timeout: 200
```
Comment (tsg) [inline docs/configuration.asciidoc]: still a dash here

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -430,6 +434,13 @@ Note:
 Values will be base64 encoded, so actual size in json document will be 4 times
 maxbytespervalue.
 
+===== transaction-timeout
```
Comment (tsg) [inline docs/configuration.asciidoc]: and a dash here

File: config/config.go [hunk 3]
```diff
@@ -44,16 +44,21 @@ type Protocols struct {
...
-	Send_request        *bool
-	Send_response       *bool
+	ProtocolCommon      `yaml:",inline"`
 	Include_authorities *bool
 	Include_additionals *bool
 }
 
 type Http struct {
-	Ports                []int
+	ProtocolCommon       `yaml:",inline"`
```
Comment (ruflin) [inline config/config.go]: Cool way to have a general config block :+1:

**Comment by ruflin** [comment]
LGTM
@urso I would suggest to squash the commits into 1?

**Comment by urso** [comment]
@ruflin history cleanup done

**Pull request merged**